src/lando/main/models/revision.py

```diff
@@ -104,6 +115,89 @@ def serialize(self) -> dict[str, Any]:
             "updated_at": self.updated_at,
         }
 
+    @property
+    def author(self):
+        """Get the full author string in "Name <Email>" format."""
+        parts = []
+        if self.author_name:
+            parts.append(self.author_name)
+        if self.author_email:
+            parts.append(f"<{self.author_email}>")
+
+        return " ".join(parts)
+
+    @property
+    def author_name(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        return metadata.get("author_name")
+
+    @property
+    def author_email(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        return metadata.get("author_email")
+
+    @property
+    def commit_message(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        return metadata.get("commit_message")
+
+    @property
+    def timestamp(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        if ts := metadata.get("timestamp"):
+            # Some codepaths (via Phabricator) have the timestamp set as an int.
+            # We make sure it's always a string.
+            return str(ts)
+        return None
```

Comment:
Feels a bit repetitive to call `_parse_metadata_from_patch` in each method. Maybe all these values should be determined at the same time, whenever `self.patch_data` is set.

Reply:
The metadata is only parsed one, and cached, as part of the `_parse_metadata_from_patch()` method https://github.com/mozilla-conduit/lando/pull/200/files#diff-6f00d8fe4977345e9360a87afcf4861324a3b6ff012d6a6a5716640b4b44c04dR153-R168

But calling it in every method makes sure we call it the first time we ever need any metadata.

Reply:
The repetitiveness aspect (i.e., calling `_parse_metadata_from_patch` in every method) was more so what I meant, not so much efficiency. I think changing `_parse_metadata_from_patch` to a property (e.g., `_parsed_metadata_from_patch`) and then using it directly in these methods would alleviate this issue.

This would become:


```python
    @property
    def timestamp(self) -> Optional[str]:
        if ts := self._parsed_metadata_from_patch.get("timestamp"):
            # Some codepaths (via Phabricator) have the timestamp set as an int.
            # We make sure it's always a string.
            return str(ts)
        return None
```

Reply:
Ah! Yeah, that's better.